### ISSUE TITLE
Clarification requested for 3.2.4 Consistent Identification

### ISSUE THREAD

**Issue by jamieherrera** [w3c/wcag#4216]
Please confirm, Is 3.2.4 intended to be specifically about consistent identification for screen reader users and the non-visual experience, or also include the visual experience of functional components? 

Much of the language of the Understanding Document linked below, including Examples and Sufficient Techniques, center around the accessible name and text alternatives. The In Brief section, however, and the[ Predictive Guideline](https://www.w3.org/WAI/WCAG22/Understanding/predictable) this SC, more broadly provide examples of where this guideline and what "consistent" means in a broader sense, could apply to people with cognitive, low vision, and motor disabilities. 

I do see that the first paragraph of the Intent section includes cognitive and screen reader considerations, but would suggest to include more language in the Understanding document to support this messaging. 

[3.2.4 Consistent Identification](https://www.w3.org/WAI/WCAG22/Understanding/consistent-identification.html):

> Components that have the [same functionality](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html#dfn-same-functionality) within a [set of Web pages](https://www.w3.org/WAI/WCAG21/Understanding/consistent-identification.html#dfn-set-of-web-pages) are identified consistently.

**Comment by A11yTea** [comment]
I think their use of the word "label" implies the visual label because in this phrase: "If identical functions have different labels (or, more generally, a different [accessible name](https://www.w3.org/TR/accname/#dfn-accessible-name)) on different Web pages, the site will be considerably more difficult to use." They go broader "more generally a different accessible name". 

And their example: "Example 7: Example of a Failure
A submit "search" button on one Web page and a "find" button on another Web page both have a field to enter a term and list topics in the Web site related to the term submitted. In this case, the buttons have the same functionality but are not labeled consistently." is about the visual label.

I think the word to focus on for this criteria is the word "identified". So start with how a component is "identified" and then check if its consistent.

**Comment by JAWS-test** [comment]
I have always understood the SC to mean both the visual labeling and the programmatic labeling (accessible name).

Incidentally, I think the phrase “more generally” is wrong in the sentence: 

> If identical functions have different labels (or, **more generally**, a different [accessible name](https://www.w3.org/TR/accname/#dfn-accessible-name)) on different Web pages

**Comment by jamieherrera** [comment]
>  both the visual labeling and the programmatic labeling (accessible name).

The term "accessible name" is not equivalent to "programmatic labeling"; it includes the programmatic label. See [Accessible name ](https://www.w3.org/TR/accname/#dfn-accessible-name) definition. 

To go back to my original question, 
"Is 3.2.4 intended to ... include the visual experience of functional components?"

What I'm asking goes beyond labels. I am asking if 3.2.4 should also apply to using visual cues consistently (which are also identifiers). example: using a chevron consistently on a tile component that has the same function. This type of consistent identification is separate from its "name". 
The SC normative text does not limit the consistency to labels, but the Understanding document seems to lean heavily on labels.

**Comment by detlevhfischer** [comment]
Hi @jamieherrera 
I can’t see the term “programmatic labelling” in the definition of _Accessible name_ you point to above. I have treated the terms as synonymous so far. Can you explain?

**Comment by jamieherrera** [comment]
@detlevhfischer I was quoting @JAWS-test not WCAG. 

I would like to get some sort of clarification through actionable change: if 3.2.4 should be limited to the way a component is labeled that needs to be stated. If it extends to non-text visual identification as well, that should be clearer. The normative text does not explicitly mention the label, just that the component is identified consistently.

**Comment by jamieherrera** [comment]
just confirming then, "accessible name" can be seen as synonymous with "visible label"?

**Comment by jamieherrera** [comment]
Just checking in... maybe @detlevhfischer to @JAWS-test can move this forward in any upcoming conversations with the larger group?

**Comment by detlevhfischer** [comment]
@jamieherrera I think it is pretty clear that this SC is not specifically about consistent identification for screen reader users and the non-visual experience. Besides text links and button text, the understanding talks about the consistency of icons as much as abut that of alternative texts. It also has examples where the same icon may have different alt to reflect different functions - ok if that maps correctly onto the respective function.

So I think the argument could be made that the consistent use of a visual identifier of function like the chevron you mention would fall under this SC. Since there is no defined failure for inconsistent use of visual indicators, and the indicator may by some be considered redundant (since alignment of tiles or other cues may visually signify that entries are buttons) I believe it this is not a clear-cut case of a failure. As so often, it will depend on the context if use. If you have a view with some tiles that have a chevron and others that don't and that use does not line up with whether or not these are interactive, a case could be made that this is inconsistent. If use differs between views but is consistent within one view, the case would be weaker. 

I don't think solving this with a proposed working group answer will bring more clarity than "it depends"..

**Comment by A11yTea** [comment]
> Please confirm, Is 3.2.4 intended to be specifically about consistent identification for screen reader users and the non-visual experience, or also include the visual experience of functional components?

Sorry, I think I missed the point of your question. Yes, I believe that the "visual experience of functional components" is included. In the "Benefits" section it states:
> When non-text content is used in a consistent way to identify components with the same functionality, people with difficulty reading text or detecting text alternatives can interact with the web without depending on text alternatives.
So this phrase is explicitly about non-text content and not dependent on the text alternative.

However, the difficult thing here is to define what does identification mean in that content. Is the chevron an identification? So let's say we have a dropdown menu that uses a chevron to indicate it is expandable on one page and then it uses a plus icon on another page. Is this a failure? Let's see what the others think.

**Comment by TestPartners** [comment]
The WCAG definition of a label  is "text or other component with a text alternative that is presented to a user to identify a component". A supplementary note says "A label is presented to all users". In my view, icons such as chevrons are therefore an identification and must be consistent.

**Comment by patrickhlauke** [comment]
based on the current example 2, it feels like at least the original understanding https://www.w3.org/WAI/WCAG22/Understanding/consistent-identification.html#examples was leaning only towards "for SR users" here

> Example 2: Check Mark
> A check mark icon functions as "approved", on one page but as "included" on another. Since they serve different functions, they have different text alternatives.

(in this example, the visual ambiguity of the icon meaning different things on different pages remains, it's only disambiguated in the text alternative)

I would be in favor of expanding the understanding to make it clear that (if?) visual consistency was also intended - and if so, removing that example altogether, as to me that's completely incompatible with the idea of visual consistency.

however, i can aksi see how this could be a slippery slope in many cases ... one of those "these call to action links are visually styled like buttons and i want to fail them because users may think they're buttons not links" (not that normal/mortal/non-technical users generally have any idea what the difference between a link or button is...it's a tappable/clickable "thing" that does something), or "all links have an underline, so can i fail the links in the navigation bar for NOT having underlines?" etc
we're not intending to be the visual design/style police here...but where to draw the line...

it's something that would, either way, require nuance and depend on context. which is not something that's easy to convey with binary pass/fail requirements...
